### PR TITLE
Focused Launch: add a resuable summary item

### LINKED ISSUE #47147: Focused Launch: add a reusable component for purchased item
When displaying _Your domain_ and _Your plan_ in the summary step, the items should share the same styling.

#### Screenshot / Video
Screenshot 2020-11-05 at 14 19 11

#### Follow up
The reusable item could be extended to support other use cases for the summary step as discussed in p1603461627005700-slack-luna
Screenshot 2020-11-05 at 14 19 11

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

This PR adds a reusable item for the summary page in focused launch. It also temporarily adds all possible usage combinations for the item for demoing purposes. These will be deleted once this PR is reviewed. 

#### Testing instructions

1. In apps/editing-toolkit run `yarn dev --sync`.
2. Sandbox testingfocusedflow.wordpress.com.
3. Visit https://testingfocusedflow.wordpress.com/wp-admin/post-new.php.
4. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.
5. You should see all combination rendered. 

Fixes #47147

### CODE REVIEW

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 1]
```diff
@@ -283,6 +295,40 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
...
+							</FocusedLaunchSummaryItem>
+							<FocusedLaunchSummaryItem isSelected>
+								<LeadingContentSide label={ 'Selected Free Plan' } />
+								<TrailingContentSide price="Free" />
+							</FocusedLaunchSummaryItem>
+							<FocusedLaunchSummaryItem readOnly>
+								<LeadingContentSide label={ 'Disabled Free Plan' } />
+								<TrailingContentSide warningNote="Not available with your domain selection" />
+							</FocusedLaunchSummaryItem>
+						</div>
```
Comment (alshakero) [inline packages/launch/src/focused-launch/summary/index.tsx:331]: All of these will be deleted after this PR.

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/**
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx:1]: If this component needs to be used outside of Focused Launch, we could look into making it a more generic "Launch" item, or even moving it to the `@automattic/onboarding` package

**Review by ciampo — COMMENTED** [review]
Excellent work! This PR actually ticks a lot of boxes and helps a lot with the next steps

I just left a couple of comments. Once they're addressed, and the sample items deleted, this PR will be good to go :)

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/style.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+@import '~@automattic/onboarding/styles/variables';
+
+$accent-blue: #117ac9;
+
+.focused-launch-summary__item {
+	border: 2px solid var( --studio-gray-5 );
+	padding: 10px 14px;
+	min-height: 64px;
+	/* stylelint-disable-next-line scales/radii */
+	border-radius: 5px;
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/focused-launch-summary-item/style.scss:11]: From the design, it looks like the border radius is `4px`. Also, we should understand why `4px` is still not in the [list of allowed border-radii](https://github.com/Automattic/wp-calypso/blob/12df4ecc07993c51c549616de6a1c93ce7d95077/.stylelintrc#L6)

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	readOnly?: boolean;
+	children: [ ReturnType< typeof LeadingContentSide >, ReturnType< typeof TrailingContentSide > ];
+} > = ( { children, isSelected = false, readOnly = false } ) => {
+	return (
+		<div
+			className={ classnames( 'focused-launch-summary__item', {
+				'is-selected': isSelected,
+				'is-readonly': readOnly,
+			} ) }
+		>
```
[the remark this answers is not in the document]
Comment (alshakero) [reply packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx:50]: I like the button idea, it is actually a button.

**Comment by alshakero** [comment]
Addressed all feedback but kept the demo elements for reviewing purposes.

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx (hunk 2 above)
Comment (ciampo) [reply packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx:1]: Sure. I'll open a new issue and add it to the Focused Launch Backlog milestone

**Comment by alshakero** [comment]
I will go ahead and merge this to continue on https://github.com/Automattic/wp-calypso/pull/47378

**Review by ciampo — APPROVED** [review]
LGTM.

I opened #47380 and #47381 to track issues that are not a blocker for this PR.

Can be merged after the demo content gets deleted

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@alshakero please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/949028). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Pull request merged**